app/models/child_theme.rb

```diff
@@ -1,6 +1,24 @@
 class ChildTheme < ActiveRecord::Base
   belongs_to :parent_theme, class_name: 'Theme'
   belongs_to :child_theme, class_name: 'Theme'
+
+  validate :child_validations
+
+  private
+
+  def child_validations
+    if ChildTheme.exists?(["parent_theme_id = ? OR child_theme_id = ?", child_theme_id, parent_theme_id])
```

Comment:
I may be wrong here but I think this should be a uniqueness validation on `parent_theme_id` and `child_theme_id`?

Reply:
Hmmm what I want to do here is to make sure the new record's `child_theme_id` value doesn't exist in `parent_theme_id` column and vice versa. In other words, we don't want to allow children to become parents, or parents become children. It's OK if a theme is parent of multiple themes and a theme is a child for multiple parents.

Reply:
Oops ignore me here